mlrun/feature_store/steps.py

```diff
@@ -26,7 +26,39 @@
 from mlrun.utils import get_in
 
 
-class FeaturesetValidator(StepToDict, MapClass):
+def get_engine(first_event):
+    if hasattr(first_event, "body"):
+        first_event = first_event.body
+    if isinstance(first_event, pd.DataFrame):
+        return "pandas"
+    return "storey"
```

Comment:
This actually doesn't necessarily mean that the engine is pandas. A storey event can contain a pandas dataframe, and I have a vague recollection that this has been done before.

Ideally this information (which engine) will be passed down the stack.

Also, pay attention that the engine isn't always storey/pandas. In serving it's called sync/async.

Reply:
Ok, since it's only used for steps under `mlrun.feature_store.steps`.